src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,81 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
-
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
-
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+import React, { useState } from 'react';
+
+type Props = {
+  onChangeQuery: (query: string) => void,
+  onChangeSelectOption: (event: string) => void,
+
+};
+
+export const TodoFilter: React.FC<Props> = ({
+  onChangeQuery = () => {},
+  onChangeSelectOption = () => {},
+}) => {
+  const [query, setQuery] = useState('');
+  const [selectOption, setSelectOption] = useState<string>();
+
+  function onChangeSetQuery(event: React.ChangeEvent<HTMLInputElement>) {
+    onChangeQuery(event.target.value);
+    setQuery(event.target.value);
+  }
+
+  function removeQuery() {
+    onChangeQuery('');
+    setQuery('');
+  }
+
+  const onChangeOnChooseOption
+  = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setSelectOption(event?.target?.value);
+    onChangeSelectOption(event?.target?.value);
```

Comment:
```suggestion
    setSelectOption(event.target.value);
    onChangeSelectOption(event.target.value);
```
Why do you use conditional chaining here?